internal/stack/compose.go

```diff
@@ -77,6 +89,11 @@ func dockerComposeBuild(ctx context.Context, options Options) error {
 }
 
 func dockerComposePull(ctx context.Context, options Options) error {
+	// Check if Docker daemon is running
+	if err := CheckDockerRunning(); err != nil {
+		return err
+	}
```

Comment:
Instead of having to check it on every operation, I think we could create a new `newComposeProvider() (*composeProvider, error)` constructor that could run the check, when initializing it in the factory, here:
https://github.com/elastic/elastic-package/blob/af5ac62d414f18c9bf442b83618f0499ca5bdc41/internal/stack/providers.go#L59

Reply:
I think the other providers (Environment and Serverless) could take advantage of this check too. IIRC these two providers run also docker (docker-compose) commands.

Just thinking that doing this check in the providers makes that elastic-package test or install commands do not run this check.
Maybe it's not needed, since all these commands will require to be the Elastic stack up... WDYT?